### PR TITLE
hack: add a test script

### PR DESCRIPTION
**Pull request by dcbw**
@squeed @danwinship mostly cribbed from ovn-kubernetes but significantly cleaned up since we don't need sudo.

### CODE REVIEW

File: hack/test-go.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+if [ -z "$PKGS" ]; then
+  # by default, test everything that's not in vendor
+  PKGS="$(go list ./... | grep -v vendor | xargs echo)"
+fi
+
+CGO_ENABLED=0 go test "${goflags[@]:+${goflags[@]}}" ${PKGS}
+retcode=$?
+
+popd
+exit $retcode
```
Comment (danwinship) [inline hack/test-go.sh]: why popd and then exit? You could just change the "pushd" to a "cd"

Comment (dcbw) [reply hack/test-go.sh]: fixed.

**Comment by dcbw** [comment]
@danwinship @squeed PTAL thanks

File: hack/test-go.sh [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+source "$(dirname "${BASH_SOURCE}")/init.sh"
```
Comment (danwinship) [inline hack/test-go.sh]: /me just noticed for the first time that you can nest `""` inside `$()` inside `""`...

**Comment by danwinship** [comment]
/lgtm
although I wouldn't object if you submitted another PR (or sniped the commitbot on this PR) to consistently use both `""` and `${}` around all variables. (What if I want my `GOPATH` to have a space in it?!)

File: hack/test-go.sh (hunk 2 above)
Comment (dcbw) [reply hack/test-go.sh]: Yeah I noticed that this weekend, I was also surprised.

**Pull request merged**